components/zigbee/time/zigbee_time.cpp

```diff
@@ -0,0 +1,81 @@
+#include "zigbee_time.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace zigbee {
+
+ZigbeeTime *zigbeeT;
+
+static void send_timesync_request() {
+	ESP_LOGI(TAG, "Requesting time from coordinator...");
+  uint16_t attributes[] = 
+  {
+      ESP_ZB_ZCL_ATTR_TIME_TIME_ID,
+  };
+  esp_zb_zcl_read_attr_cmd_t read_req;
+  read_req.address_mode = ESP_ZB_APS_ADDR_MODE_16_ENDP_PRESENT;
+  read_req.attr_field = attributes;
+  read_req.attr_number = sizeof(attributes) / sizeof(uint16_t);
+  read_req.clusterID = ESP_ZB_ZCL_CLUSTER_ID_TIME;
+  read_req.zcl_basic_cmd.dst_endpoint = 1;
+  read_req.zcl_basic_cmd.src_endpoint = 1;
+  read_req.zcl_basic_cmd.dst_addr_u.addr_short = 0x0000;  //coordinator
+  esp_zb_zcl_read_attr_cmd_req(&read_req);
```

Comment:
you need to acquire a lock first:
```suggestion
  esp_zb_lock_acquire(portMAX_DELAY);
  esp_zb_zcl_read_attr_cmd_req(&read_req);
  esp_zb_lock_release();
```

Reply:
This is introducing the following warning in the logs:
```
[15:33:32][W][component:237]: Component zigbee.time took a long time for an operation (135 ms).
[15:33:32][W][component:238]: Components should block for at most 30 ms.
```
Presumably not an issue given the infrequency of requesting time...?

Reply:
Interesting that the zigbee task blocks for so long. Never saw that with set_attribute before.
Anyway, we can easily avoid that by using something like:
```
if (esp_zb_lock_acquire(30/portTICK_PERIOD_MS) {
  esp_zb_zcl_read_attr_cmd_req(&read_req);
  esp_zb_lock_release();
  this->requested_ = true;
}
```
If it does not succeed within 30ms, we would try again with the next loop() call.